src/_guides/language/coming-from/swift-to-dart.md

```diff
@@ -0,0 +1,2982 @@
+ 'fifth': 'golden rings'
+};
+
+final nobleGases = {
+ 2: 'helium',
+ 10: 'neon',
+ 18: 'argon',
+};
+```
+
+{:.include-lang}
+```swift
+let gifts = [
+   "first": "partridge",
+   "second": "turtle doves",
+   "fifth": "golden rings",
+];
+
+let nobleGases = [
+   2: "helium",
+   10: "neon",
+   18: "argon",
+];
+```
+
+The following code samples provide an overview
+of the basic actions that you can perform on a
+Dart `Map`. The first example shows how to
+retrieve a value from a `Map` using the `key` operator:
+
+{:.include-lang}
+```dart
+final gifts = {'first': 'partridge'};
+final gift = gifts['first']; // ‘partridge'
+```
+
+Use the `containsKey` method to check whether a
+key is already present in the `Map`:
+
+{:.include-lang}
+```dart
+final gifts = {'first': 'partridge'};
+assert(gifts.containsKey('fifth')); // false
+```
+
+Use the index assignment operator (`[]=`) to add
+or update an entry in the `Map`. If the `Map`
+doesn't contain the key yet, the entry is added.
+If the key is present, the entry's value is updated:
+
+{:.include-lang}
+```dart
+final gifts = {'first': 'partridge'};
+gifts['second'] = 'turtle'; // Gets added
+gifts['second'] = 'turtle doves'; // Gets updated
+```
+
+To remove an entry from the `Map` use the `remove` method,
+and to removes all entries that satisfy a given test
+use the `removeWhere` method:
+
+{:.include-lang}
+```dart
+final gifts = {'first': 'partridge'};
+gifts.remove('first');
+gifts.removeWhere((key, value) => value == 'partridge');
+```
+
+## Classes 
+
+Dart doesn't define an interface type&mdash;_any_
+class can be used as an interface.
+If you want to introduce just an interface,
+create an abstract class with no concrete members.
+To get a more detailed understanding of these categories,
+check out the documentation in the
+[abstract classes](#abstract-classes),
+[implicit interfaces](#implicit-interfaces),
+and [extending a class](#extending-a-class) sections.
+
+Dart doesn't provide support for value types.
+As mentioned in the [Built-in types](#built-in-types) section,
+all types in Dart are reference types (even primitives),
+meaning that Dart doesn't provide a `struct` keyword.
+
+### Enums
+
+Enumerated types, often called enumerations or enums,
+are a special kind of class used to represent
+a fixed number of constant values. Enums have been
+part of the Dart language for a long time,
+but Dart 2.17 added enhanced enums support for members.
+That means you can add fields holding state,
+constructors that set that state,
+methods with functionality,
+and even override existing members.
+For more information, check out
+[Enhanced enums with members][] on the Dart blog,
+and [Declaring enhanced enums][] in the Dart language tour.
+
+[Declaring enhanced enums]: /guides/language/language-tour#declaring-enhanced-enums
+[Enhanced enums with members]: https://medium.com/dartlang/dart-2-17-b216bfc80c5d#:~:text=verbose%20and%20repetitive.-,Enhanced,-enums%20with%20members
```

Comment:
I would rather just link to the language tour as the blog may get out of date and not all browsers support the text search syntax:

```suggestion
[Declaring enhanced enums][] in the Dart language tour.

[Declaring enhanced enums]: /guides/language/language-tour#declaring-enhanced-enums
```